### PR TITLE
release-note: Describe issues around node admission in 1.22

### PR DESCRIPTION
**Pull request by smarterclayton**
The 1.22 release fixed an issue where pods that were terminating
were not always properly accounting for the resources they used.
As a consequence, certain workloads that saturate a single node with
pods may see increased pod creation failures until existing pods
fully terminate. Inform users of that change and link to where we
will resolve in the future.

#### What type of PR is this?

/kind documentation

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-1.22.md [hunk 1]
```diff
@@ -826,6 +826,10 @@ A regression bug was found where guaranteed Pods with multiple containers do not
...
...standing issue in the Kubelet where terminating pods were [vulnerable to race conditions](https://github.com/kubernetes/kubernetes/pull/102344) leading to early shutdown, resource leaks, or long delays in actually completing pod shutdown. As a consequence of this change the Kubelet now correctly takes into account the resources of running and terminating pods when deciding to accept new pods, since terminating pods are still holding on to those resources. This stricter handling may surface to end users as pod rejections when creating pods that are scheduled to mostly full nodes that have other terminating pods holding the resources the new pods need. The most likely error would be a pod set to `Failed` phase with reason set to `OutOfCpu` or `OutOfMemory`, but any resource on the node that has some fixed limit (including persistent volume counts on cloud nodes, exclusive CPU cores, or unique hardware devices) could trigger the failure. While this behavior is correct it reduces the throughput of pod execution and creates user-visible warnings - [future versions of Kubernetes will minimize the likelihood users see rejected pods](https://github.com/kubernetes/kubernetes/issues/106884).
```
Comment (alculquicondor) [inline CHANGELOG/CHANGELOG-1.22.md:831]: it's a "warning" in the case of a workload pod (replicaset, statefulset, etc). But it's a complete failure in the case of plain pods or jobs with a small `backoffLimit`.
Can we suggest users to not use plain pods or backoffLimit=0?

**Comment by dims** [comment]
/assign alculquicondor

@smarterclayton one suggestion inline from reviewer!

File: CHANGELOG/CHANGELOG-1.22.md (hunk 1 above)
Comment (smarterclayton) [reply CHANGELOG/CHANGELOG-1.22.md]: It was already a complete failure for those users during eviction anyway.  Do we have a doc that describes the recommended interactions here to reference directly, vs inlining it?  I'd probably prefer 

https://kubernetes.io/docs/concepts/scheduling-eviction/#pod-disruption is close, we could potentially describe in more detail there.

https://kubernetes.io/docs/tasks/extend-kubernetes/ is probably a good place where we could describe "writing a controller that schedules pods onto a cluster" in detail, including this warning.

Added a stub sentence pending us deciding where to put a better section.

File: CHANGELOG/CHANGELOG-1.22.md [hunk 2]
```diff
@@ -826,6 +826,10 @@ A regression bug was found where guaranteed Pods with multiple containers do not
...
...a consequence of this change the Kubelet now correctly takes into account the resources of running and terminating pods when deciding to accept new pods, since terminating pods are still holding on to those resources. This stricter handling may surface to end users as pod rejections when creating pods that are scheduled to mostly full nodes that have other terminating pods holding the resources the new pods need. The most likely error would be a pod set to `Failed` phase with reason set to `OutOfCpu` or `OutOfMemory`, but any resource on the node that has some fixed limit (including persistent volume counts on cloud nodes, exclusive CPU cores, or unique hardware devices) could trigger the failure. While this behavior is correct it reduces the throughput of pod execution and creates user-visible warnings - [future versions of Kubernetes will minimize the likelihood users see rejected pods due to this issue](https://github.com/kubernetes/kubernetes/issues/106884). In general, any automation that creates pods [must take Kubelet rejections into account](https://kubernetes.io/docs/concepts/scheduling-eviction/#pod-disruption), and should be designed to retry and backoff where necessary.
```
Comment (alculquicondor) [inline CHANGELOG/CHANGELOG-1.22.md:831]: "likelihood that users see" might read better

File: CHANGELOG/CHANGELOG-1.22.md (hunk 1 above)
Comment (alculquicondor) [reply CHANGELOG/CHANGELOG-1.22.md]: can we say "user-visible pod failures" instead?

Maybe a better link is https://kubernetes.io/docs/concepts/workloads/

Comment (smarterclayton) [reply CHANGELOG/CHANGELOG-1.22.md]: In the context of the discussion, https://kubernetes.io/docs/concepts/workloads/ doesn't seem to address taking kubelet rejection into account directly - which part do you view as important to communicate to users?

File: CHANGELOG/CHANGELOG-1.22.md (hunk 2 above)
Comment (smarterclayton) [reply CHANGELOG/CHANGELOG-1.22.md:831]: Updated

**Pull request merged**